### PR TITLE
test(insight, ad-ldap and user groups): added test cases

### PR DESCRIPTION
**Pull request by furqanmlk**
#### Summary

https://mattermost.atlassian.net/browse/MM-41906
https://mattermost.atlassian.net/browse/MM-46112
https://mattermost.atlassian.net/browse/MM-47145
https://mattermost.atlassian.net/browse/MM-48186
https://mattermost.atlassian.net/browse/MM-44347

#### Is the feature in stable branch or under review?
- [x] Feature in stable branch (master)

#### Test client
- [x] Web browser

<a href="https://gitpod.io/#https://github.com/mattermost/mattermost-test-management/pull/19"></a>

### CODE REVIEW

File: data/test-cases/suite/ad-ldap/MM-test-case-1.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "Allow admins to re-add a user to all of their default group-teams and channels"
+status: Active
+priority: Normal
+folder: suite
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/ad-ldap/MM-test-case-1.md:6]: @furqanmlk I think this should be AD-LDAP

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+# (Required) Ensure all values are filled up
+name: "Add max height and overflow to Multiselect"
+status: Active
+priority: Normal
+folder: user groups
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md:6]: @furqanmlk here I think, the folder name should be `enterprise/user-group` (not in the file, but the folder should have `-`)

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-3.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+# (Required) Ensure all values are filled up
+name: " Increase number of people that can be added to a group at once"
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-3.md:3]: just a small formatting

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
try `deno task validate` locally it should suggest folder names

File: data/test-cases/suite/ad-ldap/MM-test-case-1.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Steps:
+
+1. Login as 2 LDAP users (userA and userB) in an LDAP group.
+2. Configure the group to sync to a team and a channel.
+3. Have both users leave the channel and the team.
+4. Go to **System Console** > **Users**, go to **userA**, and from the drop down menu select **Reset group-defaulted teams and channels**.
+
+**Expected**
+step 2: the users should get added to the team and then channel.
+step 4: **userA** should be re-added to the team and the channe and **userB** should not get re-added to the team or the channel.
```
Comment (saturninoabril) [inline data/test-cases/suite/ad-ldap/MM-test-case-1.md:48]:
```suggestion
step 4: **userA** should be re-added to the team and the channel, and **userB** should not get re-added to the team or the channel.
```

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+# (Required) Ensure all values are filled up
+name: "Add max height and overflow to Multiselect"
```
Comment (saturninoabril) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md:3]: I initially thought that this title is a bit technical in nature and more on technical implementation details. Also, looking at the content (step/expectation), I feel like E2E is overkill and should've been caught in unit/component test. Thoughts?

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+1. Login as **SysAdmin**.
+2. Create a channel **ChannelA**.
+3. Create 10 users.
+4. Add 10 users in **ChannelA**.
+5. Click **+** icon on LHS.
+6. Click on **Create New User Groups**.
+7. Type **Name** in Create Group modal.
+8. Add 10 users.
+9. Click **Create Group** button.
+10. Goto **ChannelA**.
```
Comment (saturninoabril) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md:50]:
```suggestion
10. Go to **ChannelA**.
```

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+11. Type Group mention and press **enter**.
+12. Click group mention .
```
Comment (saturninoabril) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md:52]:
```suggestion
12. Click group mention.
```

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+13. Click group header.
+14. Close group model.
+15. Click group mention again and select any user.
+
+**Expected**
+step 10: Following results should be displayed:
+
+- user groups popover should be displayed.
```
Comment (saturninoabril) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-2.md:60]:
```suggestion
- User groups popover should be displayed.
```

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-4.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+1. Login as **SysAdmin**.
+2. Create 2 users.
+3. Click **+** icon on LHS.
+4. Click on **Create New User Groups**.
+5. Type **Name** "test-group" in Create Group modal.
+6. Add 2 users.
+7. Click Create Group button.
+
+**Expected**
+on step 5: "test-group" should be visiblle in **Mention**.
```
Comment (saturninoabril) [inline data/test-cases/suite/enterprise/user groups/MM-Test-Case-4.md:50]:
```suggestion
on step 5: "test-group" should be visible in **Mention**.
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @furqanmlk for adding test cases! LGTM. Left several comments but mostly formatting, typo, etc.

File: data/test-cases/suite/ad-ldap/MM-test-case-1.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+Ref: <https://mattermost.atlassian.net/browse/MM-48186>
+
+Steps:
+
+1. Login as 2 LDAP users (userA and userB) in an LDAP group.
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/ad-ldap/MM-test-case-1.md:42]: login as 2 in separate browsers ?

File: data/test-cases/suite/enterprise/user-groups/MM-Test-Case-1.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+**Step 1**
+
+Ref: <https://mattermost.atlassian.net/browse/MM-41906>
+
+Steps:
+
+1. Login as **SysAdmin**.
+2. Create 10 users.
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/enterprise/user-groups/MM-Test-Case-1.md:43]: can we add a sub test to use `/test cmd to create or should we manually invite users`

File: data/test-cases/suite/enterprise/user-groups/MM-Test-Case-2.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
+---
+# (Required) Ensure all values are filled up
+name: "Add popover for user groups"
```
Comment (yasserfaraazkhan) [inline data/test-cases/suite/enterprise/user-groups/MM-Test-Case-2.md:3]: Kind of confused with title, `add popover`

should it be, `Verify pop-over appears ...`

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]

File: data/test-cases/suite/ad-ldap/MM-test-case-1.md (hunk 10 above)
Comment (furqanmlk) [reply data/test-cases/suite/ad-ldap/MM-test-case-1.md:42]: Log in as ```2 ldap users```

File: data/test-cases/suite/enterprise/user-groups/MM-Test-Case-1.md (hunk 11 above)
Comment (furqanmlk) [reply data/test-cases/suite/enterprise/user-groups/MM-Test-Case-1.md:43]: For manual testing, it is up to the testers. Sometimes tester starts the server with ```--test-data``` or ```make test-data```.

**Review by yasserfaraazkhan — APPROVED** [review]
🌮

File: data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md (hunk 5 above)
Comment (saturninoabril) [reply data/test-cases/suite/enterprise/user groups/MM-Test-Case-1.md]: I see, thanks for checking!

File: data/test-cases/suite/activity-and-insights/insights/MM-Test-Case-1.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+# (Required) Ensure all values are filled up
+name: "My Insights license change"
+status: Active
+priority: Normal
+folder: Insights
+authors: "@furqanmlk"
+team_ownership: 
+- Suite Users
+
+# (Optional)
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply data/test-cases/suite/activity-and-insights/insights/MM-Test-Case-1.md:11]: Good point and I agree. I'll update the format as required.

**Review by saturninoabril — APPROVED** [review]
Thanks @furqanmlk!

**Comment by saturninoabril** [comment]
@furqanmlk, test cases saved - https://github.com/mattermost/mattermost-test-management/commit/ebabb55c03550c3d1ff0cf184ac7e99c1bd08d9c. Note that I've manually changed the invalid Location value for it's not present in Zephyr. Currently, we're not saving anything to Zephyr whenever there's new custom field. Instead, it should be created and managed in the Zephyr's UI itself. I've created https://github.com/mattermost/mattermost-test-management/issues/25 to add validation.

Also, I changed folder name from "user-groups" to "User Groups" for consistency.

**Pull request merged**